src/linux/PeripheralBase.h

```diff
@@ -32,12 +32,15 @@ class PeripheralBase {
     std::vector<BluetoothService> services();
     std::map<uint16_t, ByteArray> manufacturer_data();
 
-    ByteArray read(BluetoothUUID service, BluetoothUUID characteristic);
-    void write_request(BluetoothUUID service, BluetoothUUID characteristic, ByteArray data);
-    void write_command(BluetoothUUID service, BluetoothUUID characteristic, ByteArray data);
-    void notify(BluetoothUUID service, BluetoothUUID characteristic, std::function<void(ByteArray payload)> callback);
-    void indicate(BluetoothUUID service, BluetoothUUID characteristic, std::function<void(ByteArray payload)> callback);
-    void unsubscribe(BluetoothUUID service, BluetoothUUID characteristic);
+    // TODO: The ByteArrays in the following functions could probably be passed by reference as well
```

Comment:
Same as with the other one, Byte arrays should probably be const&

Also, don't forget to extend these definitions to WIndows and MacOS too, I can check that everything builds and fix if something is broken.

Reply:
Okay, I'll go over those definitions too, although I can only compile on Linux, so there's no way for me to test the changes.